api/controller/sshtunneler/client_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshtunneler
+
+import (
+	"github.com/juju/names/v5"
+	"github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	basetesting "github.com/juju/juju/api/base/testing"
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/rpc/params"
+	"github.com/juju/juju/state"
+)
+
+type sshTunnelerSuite struct {
+	testing.IsolationSuite
+}
+
+var _ = gc.Suite(&sshTunnelerSuite{})
+
+func newClient(f basetesting.APICallerFunc) (*Client, error) {
+	return NewClient(basetesting.BestVersionCaller{APICallerFunc: f, BestVersion: 1})
+}
+
+func (s *sshTunnelerSuite) TestControllerAddresses(c *gc.C) {
+	client, err := newClient(
+		func(objType string, version int, id, request string, arg, result interface{}) error {
+			c.Check(objType, gc.Equals, "SSHTunneler")
+			c.Check(id, gc.Equals, "")
+			c.Check(request, gc.Equals, "ControllerAddresses")
+			c.Assert(arg, gc.DeepEquals, params.Entity{Tag: "machine-1"})
+			c.Assert(result, gc.FitsTypeOf, &params.StringsResult{})
+
+			*(result.(*params.StringsResult)) = params.StringsResult{
+				Result: []string{"1.2.3.4"},
+			}
+			return nil
+		},
+	)
+	c.Assert(err, jc.ErrorIsNil)
+
+	entity := names.NewMachineTag("1")
```

Comment:
perhaps move that to the very beginning of the test and  use 'entity.String()' on L34 instead of the "machine-1" string.